github.com/cockroachdb/cockroach/pkg/sql: TestShowTraceReplica failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/381eb8d6d514272ca1857b24c58df68c574e3f31

Parameters:
```
TAGS=
GOFLAGS=
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=664944&tab=buildLog

```
I180518 05:47:00.211626 151628 server/server.go:782  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180518 05:47:00.214246 151628 server/config.go:539  [n?] 1 storage engine initialized
I180518 05:47:00.214289 151628 server/config.go:542  [n?] RocksDB cache size: 128 MiB
I180518 05:47:00.214320 151628 server/config.go:542  [n?] store 0: in-memory, size 0 B
I180518 05:47:00.217376 151628 server/node.go:376  [n?] **** cluster 7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59 has been created
I180518 05:47:00.217423 151628 server/server.go:1356  [n?] **** add additional nodes by specifying --join=127.0.0.1:37319
I180518 05:47:00.218169 151628 storage/store.go:1443  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180518 05:47:00.218269 151628 server/node.go:506  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=7043.00 p25=7043.00 p50=7043.00 p75=7043.00 p90=7043.00 pMax=7043.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180518 05:47:00.218349 151628 server/node.go:354  [n1] node ID 1 initialized
I180518 05:47:00.218445 151628 gossip/gossip.go:333  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:37319" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:3 > 
I180518 05:47:00.218678 151628 storage/stores.go:222  [n1] read 0 node addresses from persistent storage
I180518 05:47:00.219134 151628 server/node.go:647  [n1] connecting to gossip network to verify cluster ID...
I180518 05:47:00.219199 151628 server/node.go:672  [n1] node connected via gossip and verified as part of cluster "7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59"
I180518 05:47:00.219295 151628 server/node.go:440  [n1] node=1: started with [n1=<in-mem>] engine(s) and attributes []
I180518 05:47:00.219441 151628 server/server.go:1483  [n1] starting https server at 127.0.0.1:38919
I180518 05:47:00.219493 151628 server/server.go:1484  [n1] starting grpc/postgres server at 127.0.0.1:37319
I180518 05:47:00.219530 151628 server/server.go:1485  [n1] advertising CockroachDB node at 127.0.0.1:37319
I180518 05:47:00.220502 151696 storage/replica_command.go:863  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
W180518 05:47:00.238783 151885 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=a9f4f65b key=/Table/SystemConfigSpan/Start rw=true pri=0.01275595 iso=SERIALIZABLE stat=PENDING epo=0 ts=1526622420.224090371,0 orig=1526622420.224090371,0 max=1526622420.224090371,0 wto=false rop=false seq=7
I180518 05:47:00.241971 151870 storage/replica_command.go:863  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180518 05:47:00.242434 151879 sql/event_log.go:124  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180518 05:47:00.254579 151871 storage/replica_command.go:863  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
W180518 05:47:00.255606 151647 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=23bb9178 key=/Table/SystemConfigSpan/Start rw=true pri=0.05645831 iso=SERIALIZABLE stat=PENDING epo=0 ts=1526622420.246652999,0 orig=1526622420.246652999,0 max=1526622420.246652999,0 wto=false rop=false seq=13
I180518 05:47:00.257278 151585 sql/event_log.go:124  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:root}
I180518 05:47:00.261370 151913 storage/replica_command.go:863  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
W180518 05:47:00.290625 151551 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=135f379d key=/Table/SystemConfigSpan/Start rw=true pri=0.02169376 iso=SERIALIZABLE stat=PENDING epo=0 ts=1526622420.257619965,0 orig=1526622420.257619965,0 max=1526622420.257619965,0 wto=false rop=false seq=8
I180518 05:47:00.290930 151938 storage/replica_command.go:863  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180518 05:47:00.321662 151960 storage/replica_command.go:863  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
W180518 05:47:00.327066 151957 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=135f379d key=/Table/SystemConfigSpan/Start rw=true pri=0.02169376 iso=SERIALIZABLE stat=PENDING epo=0 ts=1526622420.257619965,0 orig=1526622420.257619965,0 max=1526622420.257619965,0 wto=false rop=false seq=8
I180518 05:47:00.329268 151895 sql/event_log.go:124  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180518 05:47:00.356297 151963 storage/replica_command.go:863  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180518 05:47:00.359291 151901 sql/event_log.go:124  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:root}
I180518 05:47:00.364154 151628 server/server.go:1562  [n1] done ensuring all necessary migrations have run
I180518 05:47:00.364205 151628 server/server.go:1565  [n1] serving sql connections
I180518 05:47:00.365432 152002 storage/replica_command.go:863  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180518 05:47:00.368137 151969 sql/event_log.go:124  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:37319} Attrs: Locality: ServerVersion:2.0-3} ClusterID:7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59 StartedAt:1526622420219222642 LastUp:1526622420219222642}
I180518 05:47:00.372447 151935 storage/replica_command.go:863  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180518 05:47:00.377599 151487 storage/replica_command.go:863  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180518 05:47:00.383732 152054 storage/replica_command.go:863  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180518 05:47:00.391513 151967 server/server_update.go:66  [n1] no need to upgrade, cluster already at the newest version
I180518 05:47:00.396460 151992 storage/replica_command.go:863  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180518 05:47:00.401809 152075 storage/replica_command.go:863  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180518 05:47:00.407699 151995 storage/replica_command.go:863  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180518 05:47:00.412960 151982 storage/replica_command.go:863  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180518 05:47:00.419147 152076 storage/replica_command.go:863  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180518 05:47:00.439848 152104 storage/replica_command.go:863  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180518 05:47:00.446714 152106 storage/replica_command.go:863  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180518 05:47:00.454166 152032 storage/replica_command.go:863  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180518 05:47:00.487318 151628 server/server.go:782  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180518 05:47:00.528353 151628 server/config.go:539  [n?] 1 storage engine initialized
I180518 05:47:00.528387 151628 server/config.go:542  [n?] RocksDB cache size: 128 MiB
I180518 05:47:00.528399 151628 server/config.go:542  [n?] store 0: in-memory, size 0 B
W180518 05:47:00.528486 151628 gossip/gossip.go:1293  [n?] no incoming or outgoing connections
I180518 05:47:00.528541 151628 server/server.go:1358  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180518 05:47:00.545110 152153 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:37319
I180518 05:47:00.545511 152236 gossip/server.go:219  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:38217}
I180518 05:47:00.546500 151628 storage/stores.go:222  [n?] read 0 node addresses from persistent storage
I180518 05:47:00.546543 151628 server/node.go:647  [n?] connecting to gossip network to verify cluster ID...
I180518 05:47:00.546570 151628 server/node.go:672  [n?] node connected via gossip and verified as part of cluster "7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59"
I180518 05:47:00.547036 152211 storage/stores.go:241  [n?] wrote 1 node addresses to persistent storage
I180518 05:47:00.548707 151628 kv/dist_sender.go:368  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180518 05:47:00.549554 151628 server/node.go:347  [n?] new node allocated ID 2
I180518 05:47:00.549663 151628 gossip/gossip.go:333  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:38217" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:3 > 
I180518 05:47:00.549758 151628 server/node.go:421  [n2] node=2: asynchronously bootstrapping engine(s) [n2=<in-mem>]
I180518 05:47:00.549791 151628 server/node.go:440  [n2] node=2: started with [] engine(s) and attributes []
I180518 05:47:00.549899 151628 server/server.go:1483  [n2] starting https server at 127.0.0.1:37619
I180518 05:47:00.549931 151628 server/server.go:1484  [n2] starting grpc/postgres server at 127.0.0.1:38217
I180518 05:47:00.549949 151628 server/server.go:1485  [n2] advertising CockroachDB node at 127.0.0.1:38217
I180518 05:47:00.550570 152237 storage/stores.go:241  [n1] wrote 1 node addresses to persistent storage
I180518 05:47:00.551550 151628 server/server.go:1562  [n2] done ensuring all necessary migrations have run
I180518 05:47:00.551611 151628 server/server.go:1565  [n2] serving sql connections
I180518 05:47:00.554124 152121 server/node.go:628  [n2] bootstrapped store [n2,s2]
I180518 05:47:00.564208 152286 sql/event_log.go:124  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:38217} Attrs: Locality: ServerVersion:2.0-3} ClusterID:7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59 StartedAt:1526622420549776443 LastUp:1526622420549776443}
I180518 05:47:00.567629 152284 server/server_update.go:66  [n2] no need to upgrade, cluster already at the newest version
I180518 05:47:00.588681 151628 server/server.go:782  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180518 05:47:00.621135 152161 storage/replica_raftstorage.go:520  [replicate,n1,s1,r8/1:/Table/1{1-2}] generated preemptive snapshot f9bcf4e1 at index 20
I180518 05:47:00.638224 151628 server/config.go:539  [n?] 1 storage engine initialized
I180518 05:47:00.638244 151628 server/config.go:542  [n?] RocksDB cache size: 128 MiB
I180518 05:47:00.638255 151628 server/config.go:542  [n?] store 0: in-memory, size 0 B
W180518 05:47:00.638309 151628 gossip/gossip.go:1293  [n?] no incoming or outgoing connections
I180518 05:47:00.638355 151628 server/server.go:1358  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180518 05:47:00.657623 152161 storage/store_snapshot.go:605  [replicate,n1,s1,r8/1:/Table/1{1-2}] streamed snapshot to (n2,s2):?: kv pairs: 9, log entries: 10, rate-limit: 2.0 MiB/sec, 2ms
I180518 05:47:00.658071 152475 storage/replica_raftstorage.go:730  [n2,s2,r8/?:{-}] applying preemptive snapshot at index 20 (id=f9bcf4e1, encoded size=6469, 1 rocksdb batches, 10 log entries)
I180518 05:47:00.658859 152475 storage/replica_raftstorage.go:736  [n2,s2,r8/?:/Table/1{1-2}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180518 05:47:00.660271 152161 storage/replica_command.go:1777  [replicate,n1,s1,r8/1:/Table/1{1-2}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r8:/Table/1{1-2} [(n1,s1):1, next=2]
I180518 05:47:00.663331 152161 storage/replica.go:3311  [n1,s1,r8/1:/Table/1{1-2}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180518 05:47:00.667020 152205 storage/raft_transport.go:459  [n2] raft transport stream to node 1 established
I180518 05:47:00.669873 152463 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:37319
I180518 05:47:00.670304 152490 gossip/server.go:219  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:34089}
I180518 05:47:00.670495 152249 storage/replica_proposal.go:202  [n2,s2,r8/2:/Table/1{1-2}] new range lease repl=(n2,s2):2 seq=3 start=1526622420.668731805,0 epo=1 pro=1526622420.668735716,0 following repl=(n1,s1):1 seq=2 start=1526622420.217974292,0 exp=1526622429.218233003,0 pro=1526622420.218255457,0
I180518 05:47:00.671468 151628 storage/stores.go:222  [n?] read 0 node addresses from persistent storage
I180518 05:47:00.671520 151628 storage/stores.go:241  [n?] wrote 2 node addresses to persistent storage
I180518 05:47:00.671538 151628 server/node.go:647  [n?] connecting to gossip network to verify cluster ID...
I180518 05:47:00.671559 151628 server/node.go:672  [n?] node connected via gossip and verified as part of cluster "7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59"
I180518 05:47:00.673024 151628 kv/dist_sender.go:368  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180518 05:47:00.673952 151628 server/node.go:347  [n?] new node allocated ID 3
I180518 05:47:00.674017 151628 gossip/gossip.go:333  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:34089" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:3 > 
I180518 05:47:00.674072 151628 server/node.go:421  [n3] node=3: asynchronously bootstrapping engine(s) [n3=<in-mem>]
I180518 05:47:00.674100 151628 server/node.go:440  [n3] node=3: started with [] engine(s) and attributes []
I180518 05:47:00.674200 151628 server/server.go:1483  [n3] starting https server at 127.0.0.1:37431
I180518 05:47:00.674216 151628 server/server.go:1484  [n3] starting grpc/postgres server at 127.0.0.1:34089
I180518 05:47:00.674229 151628 server/server.go:1485  [n3] advertising CockroachDB node at 127.0.0.1:34089
I180518 05:47:00.688897 152612 storage/stores.go:241  [n1] wrote 2 node addresses to persistent storage
I180518 05:47:00.690933 152556 storage/stores.go:241  [n2] wrote 2 node addresses to persistent storage
I180518 05:47:00.691320 151628 server/server.go:1562  [n3] done ensuring all necessary migrations have run
I180518 05:47:00.691352 151628 server/server.go:1565  [n3] serving sql connections
I180518 05:47:00.698333 152533 server/node.go:628  [n3] bootstrapped store [n3,s3]
I180518 05:47:00.705978 152377 sql/event_log.go:124  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:34089} Attrs: Locality: ServerVersion:2.0-3} ClusterID:7b4ce99b-70a5-4a7a-bbe3-eeb0de129e59 StartedAt:1526622420674078816 LastUp:1526622420674078816}
I180518 05:47:00.707570 152375 server/server_update.go:66  [n3] no need to upgrade, cluster already at the newest version
I180518 05:47:00.730758 152709 sql/event_log.go:124  [n1,client=127.0.0.1:37668,user=root] Event: "create_database", target: 50, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I180518 05:47:00.732133 152732 storage/replica_command.go:863  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180518 05:47:00.735593 152709 sql/event_log.go:124  [n1,client=127.0.0.1:37668,user=root] Event: "create_table", target: 51, info: {TableName:d.public.t1 Statement:CREATE TABLE d.public.t1 (a INT PRIMARY KEY) User:root}
I180518 05:47:00.740293 152720 storage/replica_command.go:863  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180518 05:47:00.740707 152604 storage/replica_raftstorage.go:520  [replicate,n1,s1,r20/1:/Table/{23-50}] generated preemptive snapshot 47b59e97 at index 14
I180518 05:47:00.749181 152709 sql/event_log.go:124  [n1,client=127.0.0.1:37668,user=root] Event: "create_table", target: 52, info: {TableName:d.public.t2 Statement:CREATE TABLE d.public.t2 (a INT PRIMARY KEY) User:root}
I180518 05:47:00.749846 152573 storage/replica_command.go:863  [split,n1,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180518 05:47:00.756786 152604 storage/store_snapshot.go:605  [replicate,n1,s1,r20/1:/Table/{23-50}] streamed snapshot to (n3,s3):?: kv pairs: 13, log entries: 4, rate-limit: 2.0 MiB/sec, 0ms
I180518 05:47:00.757087 152797 storage/replica_raftstorage.go:730  [n3,s3,r20/?:{-}] applying preemptive snapshot at index 14 (id=47b59e97, encoded size=2459, 1 rocksdb batches, 4 log entries)
I180518 05:47:00.757734 152797 storage/replica_raftstorage.go:736  [n3,s3,r20/?:/Table/{23-50}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I180518 05:47:00.758554 152709 sql/event_log.go:124  [n1,client=127.0.0.1:37668,user=root] Event: "create_table", target: 53, info: {TableName:d.public.t3 Statement:CREATE TABLE d.public.t3 (a INT PRIMARY KEY) User:root}
I180518 05:47:00.768167 152604 storage/replica_command.go:1777  [replicate,n1,s1,r20/1:/Table/{23-50}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r20:/Table/{23-50} [(n1,s1):1, next=2]
I180518 05:47:00.785893 152604 storage/replica.go:3311  [n1,s1,r20/1:/Table/{23-50}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I180518 05:47:00.797061 152739 storage/replica_command.go:863  [split,n1,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180518 05:47:00.797318 152709 sql/event_log.go:124  [n1,client=127.0.0.1:37668,user=root] Event: "set_zone_config", target: 51, info: {Target:d.t1 Config:constraints: [+n1] User:root}
I180518 05:47:00.800740 152844 storage/raft_transport.go:459  [n3] raft transport stream to node 1 established
W180518 05:47:00.805533 152844 storage/raft_transport.go:465  [n3] raft transport stream to node 1 failed: rpc error: code = Unavailable desc = transport is closing
I180518 05:47:00.805607 152852 util/stop/stopper.go:471  quiescing; tasks left:
1      node.Node: batch
1      [async] storage.split: processing replica
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] kv.TxnCoordSender: heartbeat loop
W180518 05:47:00.805823 152577 storage/raft_transport.go:465  [n1] raft transport stream to node 3 failed: rpc error: code = Unavailable desc = transport is closing
W180518 05:47:00.805964 152205 storage/raft_transport.go:465  [n2] raft transport stream to node 1 failed: rpc error: code = Unavailable desc = transport is closing
I180518 05:47:00.806204 152852 util/stop/stopper.go:471  quiescing; tasks left:
1      node.Node: batch
1      [async] storage.split: processing replica
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
W180518 05:47:00.806388 152477 storage/raft_transport.go:465  [n1] raft transport stream to node 2 failed: rpc error: code = Canceled desc = grpc: the client connection is closing
W180518 05:47:00.806920 151787 storage/store.go:1645  [n1,s1,r23/1:/Table/5{2-3}] unable to gossip on capacity change: node unavailable; try another peer
W180518 05:47:00.807296 152739 storage/replica.go:2823  [n1,s1,r23/1:/Table/5{2-3}] during async intent resolution: node unavailable; try another peer
I180518 05:47:00.807403 152852 util/stop/stopper.go:471  quiescing; tasks left:
1      [async] storage.replicate: processing replica
1      [async] storage.pendingLeaseRequest: requesting lease
I180518 05:47:00.809634 152852 util/stop/stopper.go:471  quiescing; tasks left:
1      [async] storage.replicate: processing replica
E180518 05:47:00.809727 152745 storage/queue.go:778  [replicate,n1,s1,r20/1:/Table/{23-50}] [n1,s1,r20/1:/Table/{23-50}]: unable to transfer lease to s3: result is ambiguous (server shutdown)
	show_trace_replica_test.go:63: error executing 'ALTER TABLE d.t2 EXPERIMENTAL CONFIGURE ZONE 'constraints: [+n2]'': pq: constraint "+n2" matches no existing nodes within the cluster - did you enter it correctly?
```

--- a/pkg/ccl/partitionccl/partition_test.go
+++ b/pkg/ccl/partitionccl/partition_test.go
@@ -23,7 +23,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/config"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
-	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/parser"
@@ -1121,28 +1120,7 @@ func setupPartitioningTestCluster(ctx context.Context, t testing.TB) (*sqlutils.
 
 	// Make sure all stores are present in the NodeStatus endpoint or else zone
 	// config changes may flake (#25488).
-	testutils.SucceedsSoon(t, func() error {
-		url := tc.Server(0).ServingAddr()
-		conn, err := tc.Server(0).RPCContext().GRPCDial(url).Connect(context.Background())
-		if err != nil {
-			return err
-		}
-		client := serverpb.NewStatusClient(conn)
-		response, err := client.Nodes(context.Background(), &serverpb.NodesRequest{})
-		if err != nil {
-			return err
-		}
-
-		if len(response.Nodes) != 3 {
-			return fmt.Errorf("not enough nodes registered: %+v", response)
-		}
-		for _, node := range response.Nodes {
-			if len(node.StoreStatuses) != 1 {
-				return fmt.Errorf("expected 1 StoreStatus in NodeStatus, got %+v", node)
-			}
-		}
-		return nil
-	})
+	tc.WaitForNodeStatuses(t)
 
 	return sqlDB, func() {
 		tc.Stopper().Stop(context.Background())

--- a/pkg/sql/show_trace_replica_test.go
+++ b/pkg/sql/show_trace_replica_test.go
@@ -53,8 +53,12 @@ func TestShowTraceReplica(t *testing.T) {
 	}}
 	tc := testcluster.StartTestCluster(t, numNodes, tcArgs)
 	defer tc.Stopper().Stop(ctx)
-	sqlDB := sqlutils.MakeSQLRunner(tc.Conns[0])
 
+	// Make sure all stores are present in the NodeStatus endpoint or else zone
+	// config changes may flake (#25488).
+	tc.WaitForNodeStatuses(t)
+
+	sqlDB := sqlutils.MakeSQLRunner(tc.Conns[0])
 	sqlDB.Exec(t, `CREATE DATABASE d`)
 	sqlDB.Exec(t, `CREATE TABLE d.t1 (a INT PRIMARY KEY)`)
 	sqlDB.Exec(t, `CREATE TABLE d.t2 (a INT PRIMARY KEY)`)

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -17,6 +17,7 @@ package testcluster
 import (
 	"context"
 	gosql "database/sql"
+	"fmt"
 	"sync"
 	"testing"
 	"time"
@@ -29,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
@@ -545,6 +547,33 @@ func (tc *TestCluster) WaitForFullReplication() error {
 		}
 	}
 	return nil
+}
+
+// WaitForNodeStatuses waits until a NodeStatus is persisted for every node and
+// store in the cluster.
+func (tc *TestCluster) WaitForNodeStatuses(t testing.TB) {
+	testutils.SucceedsSoon(t, func() error {
+		url := tc.Server(0).ServingAddr()
+		conn, err := tc.Server(0).RPCContext().GRPCDial(url).Connect(context.Background())
+		if err != nil {
+			return err
+		}
+		client := serverpb.NewStatusClient(conn)
+		response, err := client.Nodes(context.Background(), &serverpb.NodesRequest{})
+		if err != nil {
+			return err
+		}
+
+		if len(response.Nodes) != tc.NumServers() {
+			return fmt.Errorf("expected %d nodes registered, got %+v", tc.NumServers(), response)
+		}
+		for _, node := range response.Nodes {
+			if len(node.StoreStatuses) == 0 {
+				return fmt.Errorf("missing StoreStatuses in NodeStatus: %+v", node)
+			}
+		}
+		return nil
+	})
 }
 
 type testClusterFactoryImpl struct{}

